src/test/playwright/support/pageobjects/exam/ExamParticipationActions.ts

```diff
@@ -0,0 +1,91 @@
+import { Page, expect } from '@playwright/test';
+import { Fixtures } from '../../../fixtures/fixtures';
+import { Commands } from '../../commands';
+import { getExercise } from '../../utils';
+import { Dayjs } from 'dayjs';
+
+export class ExamParticipationActions {
+    protected readonly page: Page;
+
+    constructor(page: Page) {
+        this.page = page;
+    }
+
+    async selectExerciseOnOverview(index: number) {
+        await this.page.locator(`.exercise-table tr:nth-child(${index}) a`).click();
+    }
+
+    async clickSaveAndContinue() {
+        await this.page.click('#save');
+    }
+
+    async checkExerciseTitle(exerciseID: number, title: string) {
+        const exercise = getExercise(this.page, exerciseID);
+        await expect(exercise.locator('.exercise-title')).toContainText(title);
+    }
+
+    async checkExamTitle(title: string) {
+        await expect(this.page.locator('#exam-title')).toContainText(title);
+    }
+
+    async getResultScore(exerciseID?: number) {
+        const parentComponent = exerciseID ? getExercise(this.page, exerciseID) : this.page;
+        const resultScoreLocator = parentComponent.locator('#exercise-result-score');
+        await Commands.reloadUntilFound(this.page, resultScoreLocator);
+        return resultScoreLocator;
+    }
+
+    async checkResultScore(scoreText: string, exerciseID?: number) {
+        const scoreElement = await this.getResultScore(exerciseID);
+        await expect(scoreElement.getByText(new RegExp(scoreText))).toBeVisible();
+    }
+
+    async checkExamFinishedTitle(title: string) {
+        await expect(this.page.locator('#exam-finished-title')).toContainText(title, { timeout: 40000 });
+    }
+
+    async checkExamFullnameInputExists() {
+        await expect(this.page.locator('#fullname')).toBeVisible({ timeout: 30000 });
+    }
+
+    async checkYourFullname(name: string) {
+        await expect(this.page.locator('#your-name')).toContainText(name, { timeout: 30000 });
+    }
```

Comment:
Why are the timeouts so long? This would be 40/30 seconds, right?

Reply:
yes, please fix it in another PR